### DISCUSSION TITLE
MO-File for Translation compiled not correct

### DISCUSSION THREAD

**Discussion by MichaelHarmsTIB** [ckan/ckan#7054] [Q&A]
**CKAN version**
2.9.2
**Describe the bug**
I followed the instruction on:
https://docs.ckan.org/en/2.9/contributing/i18n.html
My locale is **de**
I created a new Plugin with a po-file. I wrote msgid for english and msgstr for german.
Example:
```
# German translations for ckanext-fid-translation.
# Copyright (C) 2022 ORGANIZATION
# This file is distributed under the same license as the
# ckanext-fid-translation
# project.
# FIRST AUTHOR <EMAIL@ADDRESS>, 2022.
#
msgid ""
msgstr ""
"Project-Id-Version: ckanext-fid-translation 0.0.1\n"
"Report-Msgid-Bugs-To: EMAIL@ADDRESS\n"
"POT-Creation-Date: 2022-08-29 14:57+0200\n"
"PO-Revision-Date: 2022-08-29 17:02+0200\n"
"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
"Language: de\n"
"Language-Team: de <LL@li.org>\n"
"Plural-Forms: nplurals=2; plural=(n != 1)\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=utf-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Generated-By: Babel 2.7.0\n"
msgid "Complete"
msgstr "Fertig"
```
My Babel version is 2.10.3.
Then:
python setup.py compile_catalog --locale de
But with this Babel version only the header of *.mo is generated.
Before I compiled my *.po and I got a full *.mo file. But translation doesn't work.
In my layout I uesed to this syntax: {% trans %} Datenschutzerklärung {% endtrans %}

What can be wrong?

**Expected behavior**
Translation for all my *.po messages works fine.

**Further Steps**
I manged to compile a part of my *.po file.
Actually I got 90% and this message:
catalog ckan/i18n/de/LC_MESSAGES/ckan.po is marked as fuzzy, skipping
How is the syntax for html messages?
I try another thing, I add parameter -f for fuzzy I got this:
`python setup.py compile_catalog -f --locale de
running compile_catalog
985 of 1011 messages (97%) translated in ckan/i18n/de/LC_MESSAGES/ckan.po
compiling catalog ckan/i18n/de/LC_MESSAGES/ckan.po to ckan/i18n/de/LC_MESSAGES/ckan.mo

**97% with the original *.po file. Can me someone explain why I got no 100% ?**

**Comment by MichaelHarmsTIB** [comment]
I managed to compile without errors.
But my example here:
` <li class="list-inline-item normal"><a href="{{ h.url_for('/datenschutz') }}">{% trans %}Datenschutzerklärung{% endtrans %}</a></li>`
and in po:
```
#: ckan/templates/footer.html:25
msgid "Privacy Statement"
msgstr "Datenschutzerklärung"`
Is not working, no translation.
After I updated the catalog:
python setup.py update_catalog --locale de
Uwsgi-Server doesn't work, and spite out:
/usr/lib/ckan/default/src/ckan/ckan/i18n/de/LC_MESSAGES/ckan.po (line 2765): unknown keyword instance
Looking at that adress:
`#: ckan/templates/admin/config.html:52
#, fuzzy, python-format
#| msgid "" " <p><strong>Site Title:</strong> This is the title of this CKAN
#| instance " "It appears in various places throughout CKAN.</p> "
#| "<p><strong>Style:</strong> Choose from a list of simple variations of the"
#| " main colour scheme to get a very quick custom theme working.</p> "
#| "<p><strong>Site Tag Logo:</strong> This is the logo that appears in the "
#| "header of all the CKAN instance templates.</p> <p><strong>About:</strong>"
#| " This text will appear on this CKAN instances <a "
#| "href=\"%(about_url)s\">about page</a>.</p> <p><strong>Intro "
#| "Text:</strong> This text will appear on this CKAN instances <a "
#| "href=\"%(home_url)s\">home page</a> as a welcome to visitors.</p> "
#| "<p><strong>Custom CSS:</strong> This is a block of CSS that appears in "
#| "<code>&lt;head&gt;</code> tag of every page. If you wish to customize the"
#| " templates more fully we recommend <a href=\"%(docs_url)s\" "
#| "target=\"_blank\">reading the documentation</a>.</p> "
#| "<p><strong>Homepage:</strong> This is for choosing a predefined layout "
#| "for the modules that appear on your homepage.</p> "
``
Strange, I never wrote such lines!
And my custom translation has this:
#~ "my text"
Can anyone explain me this?

**Comment by MichaelHarmsTIB** [comment]
What means:
`ckan.po:2158: positional format placeholders are unbalanced` ?

**Comment by Zharktas** [comment]
You should be following extension translation guide and not the core contribution guide https://docs.ckan.org/en/2.9/extensions/translating-extensions.html

You also should generate the translation files using `extract_messages` instead of writing them manually, by the errors you are receiving, something is broken somewhere in your templates or the translation files. I'm converting this to a discussion as this seems to be not a ckan bug itself.

**Comment by MichaelHarmsTIB** [comment]
I followed the guide, I tried it with extract messages and insert my custom translations. Compiled to 100% but there was after activating the plugin no change on the ckan website.

**Comment by Zharktas** [comment]
What ever you were translating apperared in the generated .po file ? And if you added german in the html itself, it should appear on the page regardless if your translations work, ckan is translated from english to other languages.

**Comment by MichaelHarmsTIB** [comment]
There is a *.mo and *.pot file generated, I have a po and that contains my translations. Looks like the generation is completed but no translations appears in my Footer.html. Footer is part of page.html.

**Comment by Zharktas** [comment]
Did you add ITranslation interface to plugin ? Its required for translations to work in plugins.

**Comment by MichaelHarmsTIB** [comment]
I did it, but no effect.^
I used this Docs:
https://docs.ckan.org/en/2.9/extensions/translating-extensions.html
